docs/Server-Methods.md

```diff
@@ -107,6 +107,22 @@ Function to add middlewares to Fastify, check [here](https://github.com/fastify/
 #### addHook
 Function to add a specific hook in the lifecycle of Fastify, check [here](https://github.com/fastify/fastify/blob/master/docs/Hooks.md).
 
+<a name="base-path"></a>
+#### basepath
+The full path that will be prefixed to a route.
+
+Example:
+
+```js
+fastify.register(function (instance, opts, next) {
+  instance.get('/foo', function (request, reply) {
+    // Will log "basePath: /v1"
+    request.log.info('basePath: %s', instance.basePath)
+    reply.send({basePath: instance.basePath})
+  })
+}, { prefix: '/v1' })
```

Comment:
It would be helpful to add another `register` level to show what the `basePath` will be inside multiple nested plugins with a prefix.

Also missing the `next()` call here.